src/shared/hooks/useBasket.ts

```diff
@@ -0,0 +1,63 @@
+import { useMemo, useCallback, SetStateAction, Dispatch } from 'react';
+
+import useLocalStorage from './useLocalStorage';
+
+import { Namespace } from '../types/namespaces';
+
+export type Basket = {
+  [Namespace.uniprotkb]: Set<string>;
+  [Namespace.uniref]: Set<`UniRef${50 | 90 | 100}_${string}`>;
+  [Namespace.uniparc]: Set<`UPI${string}`>;
+};
+
+type StringifiableBasket = {
+  [Namespace.uniprotkb]: Array<string>;
+  [Namespace.uniref]: Array<`UniRef${50 | 90 | 100}_${string}`>;
+  [Namespace.uniparc]: Array<`UPI${string}`>;
+};
+
+const deserialise = (stringifiableBasket: StringifiableBasket): Basket => {
+  const entries = Object.entries(stringifiableBasket);
+  const object = Object.fromEntries(
+    entries.map(([key, value]) => [key, new Set(value)])
+  ) as Basket;
+  return object;
+};
+
+const serialise = (basket: Basket): StringifiableBasket => {
+  const entries = Object.entries(basket);
+  const object = Object.fromEntries(
+    entries.map(([key, value]) => [key, Array.from(value)])
```

Comment:
Same here, there shouldn't be empty/null/undefined values, but might be worth checking?

Reply:
done